### PR TITLE
feat: replace create-react-app with degit [EXT-3404]

### PR DESCRIPTION
**Pull request by andipaetzold**
This PR removes create-react-app as dependency and clones the template using `degit`.

https://github.com/contentful/apps/pull/735 must be merged first, so the path can be adjusted

With this PR I switched to ES modules to make the imports prettier.

There will be a follow-up cleanup PR that adjusts CI and removes the old template: https://github.com/contentful/create-contentful-app/pull/846

## TODO

- [x] Adjust the name in `package.json`

### CODE REVIEW

File: packages/contentful--create-contentful-app/package.json [hunk 1]
```diff
@@ -36,10 +37,8 @@
     "@contentful/cra-template-create-contentful-app": "file:../contentful--cra-template-create-contentful-app",
```
Comment (shikaan) [inline packages/contentful--create-contentful-app/package.json:37]: we might need to remove this

File: packages/contentful--create-contentful-app/lib/index.js [hunk 2]
```diff
@@ -2,54 +2,56 @@
...
+    if (e.code === 'DEST_NOT_EMPTY') {
+      message = 'destination directory is not empty';
+    }
+    throw new Error(message);
+  }
 }
 
-function initProject() {
+async function initProject() {
+  const appFolder = process.argv[3];
```
Comment (shikaan) [inline packages/contentful--create-contentful-app/lib/index.js:52]: should we maybe validate if this is a correct path-like string?

File: packages/contentful--create-contentful-app/lib/index.js [hunk 3]
```diff
@@ -60,29 +62,25 @@ function initProject() {
...
-        onSuccess(appFolder);
-      } else {
-        process.exit(exitCode);
-      }
-    });
+    rmIfExists(resolve(fullAppFolder, 'package-lock.json'));
+    rmIfExists(resolve(fullAppFolder, 'yarn.lock'));
+    updatePackageName(fullAppFolder);
+
+    await exec('npm', ['install'], { cwd: fullAppFolder });
```
Comment (shikaan) [inline packages/contentful--create-contentful-app/lib/index.js:76]: nit: maybe we can check if npm version is >=6 as we require in the package json? I am not sure if we get any warnings or anything when executed as child process

**Review by shikaan — COMMENTED** [review]
LGTM, few minor comments

File: packages/contentful--create-contentful-app/package.json (hunk 1 above)
Comment (andipaetzold) [reply packages/contentful--create-contentful-app/package.json:37]: Yes forgot about that and did it in the follow-up PR: https://github.com/contentful/create-contentful-app/pull/846/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L52

File: packages/contentful--create-contentful-app/lib/index.js (hunk 3 above)
Comment (andipaetzold) [reply packages/contentful--create-contentful-app/lib/index.js:76]: The `engines` property in package.json of `create-contentful-app` should show a warning like this:

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'create-contentful-app@1.0.0',
npm WARN EBADENGINE   required: { node: '>=16.0.0' },
npm WARN EBADENGINE   current: { node: 'v14.15.0', npm: '7.5.3' }
npm WARN EBADENGINE }
```

File: packages/contentful--create-contentful-app/lib/index.js (hunk 2 above)
Comment (andipaetzold) [reply packages/contentful--create-contentful-app/lib/index.js:52]: What is a path-like string?

Should `path.resolve(...)` handle all these cases and just return a proper absolute path?

File: packages/contentful--create-contentful-app/lib/index.js (hunk 3 above)
Comment (shikaan) [reply packages/contentful--create-contentful-app/lib/index.js:73]: Yeah, I was more thinking about an explicit check for wen we incorporate yarn, for example. We can handle that then though

Comment (andipaetzold) [reply packages/contentful--create-contentful-app/lib/index.js:73]: Yes, we might want to check whether `yarn` is installed. And even when using `yarn`, we probably still have a minimum npm version.
Let's do this in a future PR.

File: packages/contentful--create-contentful-app/lib/index.js (hunk 2 above)
Comment (shikaan) [reply packages/contentful--create-contentful-app/lib/index.js:52]: I don't think we get any validation there

**Node REPL**
```
path.resolve(process.cwd(), '>>><>?||')
'$ABSOLUTE_PATH/>>><>?||'
```

But on Windows, for example, `\ / : * ? " < > |` are invalid chars for filenames. Maybe it's handled in the windows version of Node (and therefore `path`)

**Review by shikaan — APPROVED** [review]

**Pull request merged**